contracts/BuyBackBurner.sol

```diff
@@ -143,4 +169,40 @@ contract BuyBackBurner {
 
         require(deviation <= MAX_ALLOWED_DEVIATION, "Price deviation too high");
     }
+
+    /// @dev Bridges OLAS to Ethereum mainnet for burn.
+    /// @notice if nativeTokenAmount is zero or above the balance, it will be adjusted to current native token balance.
+    /// @param nativeTokenAmount Suggested native token amount.
+    /// @param tokenGasLimit Token gas limit for bridging OLAS to L1.
+    /// @param bridgePayload Optional additional bridge payload.
+    function sendToHigherHeights(
+        uint256 nativeTokenAmount,
+        uint256 tokenGasLimit,
+        bytes memory bridgePayload
+    ) external payable {
+        require(_locked == 1, "Reentrancy guard");
+        _locked = 2;
+
+        // Buy OLAS
+        uint256 olasAmount = _buyOLAS(nativeTokenAmount);
```

Comment:
Same implementation as we had before, just adjusted this function.

Reply:
See below. We need to logically split this into the bridge part and the swap part.